crates/ursa-tracker/src/main.rs

```diff
@@ -0,0 +1,228 @@
+    );
+    let json = json!(entry);
+
+    info!("Storing node {} with config {:?}", id, entry);
+
+    match db
+        .0
+        .insert(id.to_base58().as_bytes(), json.to_string().as_bytes())
+    {
+        Ok(_) => (StatusCode::OK, Json(json)),
+        Err(e) => {
+            tracing::error!("Error writing to db: {}", e);
+            (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                Json(json!(e.to_string())),
+            )
+        }
+    }
+}
+
+/// Prometheus HTTP Service Discovery
+async fn http_sd_handler(db: Extension<Arc<Db>>) -> (StatusCode, Json<Value>) {
+    let services: Vec<PrometheusDiscoveryChunk> =
+        db.0.iter()
+            .filter_map(|i| {
+                if let Ok((_, v)) = i {
+                    let node: Node = serde_json::from_slice(&v.as_ref()).unwrap();
+                    if !node.telemetry {
+                        return None;
+                    }
+                    Some(node.into())
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+    (StatusCode::OK, Json(json!(services)))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use libp2p::PeerId;
+
+    fn tracer() {
+        tracing_subscriber::registry()
+            .with(tracing_subscriber::EnvFilter::new(
+                std::env::var("RUST_LOG").unwrap_or_else(|_| "info".into()),
+            ))
+            .init();
+    }
+
+    fn db() -> Arc<Db> {
+        Arc::new(sled::open("tracker_db").unwrap())
+    }
+
+    async fn make_announcement(
+        db: Arc<Db>,
+        addr: Option<String>,
+        id: PeerId,
+    ) -> (StatusCode, Json<Value>) {
+        let data = TrackerRegistration {
+            id,
+            addr,
+            storage: 0,
+            p2p_port: Some(6009),
+            rpc_port: Some(6009),
+            metrics_port: Some(6009),
+            agent: "".to_string(),
+        };
+        announcement_handler(
+            HeaderMap::new(),
+            ConnectInfo(SocketAddr::from(([127, 0, 0, 1], 6969))),
+            Extension(db),
+            Extension(env::var("IPINFO_TOKEN").expect("IPINFO_TOKEN is not set")),
+            Json(data),
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn local_node_announcement() {
+        tracer();
+        let db = db();
+        let id = PeerId::random();
+
+        let res = make_announcement(db.clone(), None, id).await;
+        info!("{:?}", res);
+        assert_eq!(res.0, 200);
+
+        db.remove(id.to_string().as_bytes()).unwrap();
+    }
+
+    #[tokio::test]
+    async fn remote_node_announcement() {
+        tracer();
+        let db = db();
+        let id = PeerId::random();
+
+        let res = make_announcement(db.clone(), Some("8.8.8.8".to_string()), id).await;
+        info!("{:?}", res);
+        assert_eq!(res.0, 200);
+        db.remove(id.to_string().as_bytes()).unwrap();
+    }
+
+    // doesn't work with ipinfo
```

Comment:
Dead code

Reply:
Actually need to restore this functionality, will push soon

Reply:
Do not dead code? Or did you forget to push it?

Reply:
When i refactored the ip info api from `ip-api.com` -> `ipinfo.io`, the functionality for fetching dns was lost, glad you caught it as I still needed to provide an alternative for the new api

Reply:
https://github.com/fleek-network/ursa/pull/81/commits/e639fe86f04b9d57cca0c7748673ea79689f4332